src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -251,8 +285,10 @@ public void run() {
           Thread.sleep(rcaStateCheckIntervalMillis - duration);
         }
       } catch (InterruptedException ie) {
-        LOG.error("RCA controller thread was interrupted. Reason: {}", ie.getMessage());
-        LOG.error(ie);
+        if (!deliberateInterrupt) {
```

Comment:
Looks like this was missed in the original code as well, but we should also propagate the interrupted exception if we caught it by calling Thread.currentThread().interrupt() explicitly here.